ci/build.ps1

```diff
@@ -0,0 +1,33 @@
+param(
+    [Parameter()]
+    [ValidateSet("Debug", "Release")]
+    [string] $BuildType = "Release"
+)
+
+$SourceDirectory = (Convert-Path "$PSScriptRoot/../").Replace("\", "/")
+$WorkingDirectory = $pwd.Path
+
+$commands = @()
+if (!(Get-Command cl)) {
+    $commands += '"C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Auxiliary/Build/vcvars64.bat"'
+}
+
+$commands += @"
+cmake.exe
+    -G "Ninja"
+    -S "$SourceDirectory"
+    -B "$WorkingDirectory"
+    -DCMAKE_BUILD_TYPE:STRING=$BuildType
+    -DCMAKE_INSTALL_PREFIX:PATH="$SourceDirectory/out/install/$BuildType"
+    -DDISABLE_PYTHON_BINDINGS=1 `
+    2>&1
+"@.Replace("`r`n", "")
+
+$commands += @"
+cmake.exe --build $WorkingDirectory --config $BuildType
+"@
+
+$commands += @"
+cmake.exe --install $WorkingDirectory
+"@
+cmd /c ($commands -join " && ")
```

Comment:
Don't we use it as part of the CI process?

Reply:
Not on Cirrus. Cirrus uses the {prepare,build,test}.sh files. That said, we could definitely convert over. I have a note here to convert those files to powershell scripts once everything's been merged.

Reply:
> I made a quick pass at it for a first round of comments.
> 
> One general comment is that the code flips back and forth between `#if defined()` and `#ifdef` quite a bit. I know it's sort of superfluous, but it'd be nice to standardize on one of them. That might be something to consider for after this lands though, since it's most likely a problem across the entire code base.

Went over the calls to `#defined (_MSC_VER)` and changed them to `#ifdef _MSC_VER`

Reply:
Oh wait, this file was deleted at some point. I must have been looking at a specific commit when i added the comment. Ignore me here. 😅